static/app/utils/withGlobalSelection.tsx

```diff
@@ -9,48 +10,28 @@ type InjectedGlobalSelectionProps = {
   isGlobalSelectionReady?: boolean;
 };
 
-type State = {
-  selection: GlobalSelection;
-  isReady?: boolean;
-};
-
 /**
  * Higher order component that uses GlobalSelectionStore and provides the
  * active project
  */
 function withGlobalSelection<P extends InjectedGlobalSelectionProps>(
   WrappedComponent: React.ComponentType<P>
 ) {
-  class WithGlobalSelection extends React.Component<
-    Omit<P, keyof InjectedGlobalSelectionProps> & Partial<InjectedGlobalSelectionProps>,
-    State
-  > {
-    static displayName = `withGlobalSelection(${getDisplayName(WrappedComponent)})`;
-
-    state = GlobalSelectionStore.getState();
-
-    componentWillUnmount() {
-      this.unsubscribe();
-    }
-
-    unsubscribe = GlobalSelectionStore.listen((selection: State) => {
-      if (this.state !== selection) {
-        this.setState(selection);
-      }
-    }, undefined);
-
-    render() {
-      const {isReady, selection} = this.state;
-
-      return (
-        <WrappedComponent
-          selection={selection as GlobalSelection}
-          isGlobalSelectionReady={isReady}
-          {...(this.props as P)}
-        />
-      );
-    }
-  }
+  type Props = Omit<P, keyof InjectedGlobalSelectionProps> & InjectedGlobalSelectionProps;
+
+  const WithGlobalSelection: React.FC<Props> = props => {
+    const {selection, isReady} = useLegacyStore(GlobalSelectionStore);
+
+    const selectionProps = {
+      selection,
+      isGlobalSelectionReady: isReady,
+    };
+
+    return <WrappedComponent {...selectionProps} {...(props as P)} />;
```

Comment:
```suggestion
    return <WrappedComponent selection={selection}  isGlobalSelectionReady={isReady} {...(props as P)} />;
```

Reply:
I kinda like that prettier doesn't wrap it right now haha 😆